api/agent/agent.go

```diff
@@ -630,42 +635,38 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 	state.UpdateState(ctx, ContainerStateStart, call.slots)
 	defer state.UpdateState(ctx, ContainerStateDone, call.slots)
 
-	cid := id.New().String()
-
-	// set up the stderr to capture any logs before the slot is executed and
-	// between hot functions
-	stderr := newLineWriter(&logWriter{
-		logrus.WithFields(logrus.Fields{"between_log": true, "app_name": call.AppName, "path": call.Path, "image": call.Image, "container_id": cid}),
-	})
```

Comment:
removing this does mean we'll lose logs between calls, where this will put them in the server logs for users at least. a quick `stderr.swap(betweenLogger)` below would do.

Reply:
this one is a policy discussion for sure.. so, here, we actually block stderr & stdout, so I don't think we lose logs. (I do need to verify this claim.) A bit related to this is the fact that we freeze the containers in between calls.... (However, freeze behavior is an operator option, which could be disabled.) Given this:

1) is it OK to block stderr? Probably, since we do block stdout....
2) or keep it open, but incur I/O be noisy and log it? We could clamp this with some max, etc. too.

I'm ok with either...

Reply:
that's true, freeze reduces the surface area. one case where we would lose logs I think is if the container starts, logs and exits before the container's first slot is acquired. 

otherwise, it does seem odd to get logs from before a call was started, as a user in certain cases I could see how it would be useful (why did this fail? oh, that's why) but also confusing (why is this log line 3 seconds before the call was started? I can ignore it I guess since everything was okay? is time a circle where all roads lead back to this function call?).

am not sure what the other FaaS providers are doing here.

for stdout, it seems like we should be discarding between functions possibly, or maybe just logging it too? otherwise we can have a container that for 1 input (or 0 input) produces an additional output -- in the 1->2 case, we'd get succeeding functions that reply without accounting for the input (smells funny). it seems in general like a case of 'containers behaving badly', not a clear best way to defend here I can see. we could add it to our shut down shit list, too.